File: test/data/typescript/test02/ConsoleOutControllee.ts
import {IControllee} from "./IControllee"
export class ConsoleOutControllee implements IControllee
{
    public DoAction1(): void
    {
        console.log("Action1");
    }
    public DoAction3(): void
    {
        console.log("Action3");
    }
    public NoTransition(state: string, transition: string): void
    {
        console.log(`NoTransition(${state}: ${transition})`);
    }
    public OverTransition(transition: string): void
    {
        console.log(`OverTransition(${transition})`);
    }
}

File: test/data/cpp/test03/ConsoleOutControllee.cpp
#include "ConsoleOutControllee.h"
#include <stdio.h>
ConsoleOutControllee::ConsoleOutControllee()
{
}
ConsoleOutControllee::~ConsoleOutControllee()
{
}
void ConsoleOutControllee::DoAction0()
{
    printf("Action0\n");
}
void ConsoleOutControllee::DoAction1()
{
    printf("Action1\n");
}
void ConsoleOutControllee::DoAction2()
{
    printf("Action2\n");
}
void ConsoleOutControllee::DoAction3()
{
    printf("Action3\n");
}
void ConsoleOutControllee::NoTransition(const char* state, const char* transition)
{
    printf("NoTransition(%s: %s)\n", state, transition);
}
void ConsoleOutControllee::OverTransition(const char* transition)
{
    printf("OverTransition(%s)\n", transition);
}